### PR TITLE
🔥 Fix import added too early

### PR DESCRIPTION
**Pull request by browniebroke**
In case of module with a docstring or with a `__future__` import, the `Optional` import should be added after.

### CODE REVIEW

**Comment by Luttik** [comment]
Hi @browniebroke,

Thank you for your contibution.

I agree with the description of the pull-request. Could you add tests that validate this behavior though? That would be much appreciated!

**Comment by browniebroke** [comment]
I'm confused, I've updated the yaml file with 3 new test cases, they were failing before I changed the code and now they pass. Do I miss anything?

**Comment by Luttik** [comment]
Nope I missed something 🙈 it was really late. Ill give it a good look later today.

File: src/auto_optional/code_processing.py
Comment (Luttik) [inline src/auto_optional/code_processing.py:144]: Could you extract this into a helper method? something descriptive like `__find_body_index_for_option_import_insertion`?

**Review by Luttik — CHANGES_REQUESTED** [review]
Hi @browniebroke,
Thanks again for this contribution. I have a quite minor change that I'd like to see in this review. Just to improve the readability a bit. After that I'll merge and generate a release.

File: src/auto_optional/code_processing.py
Comment (browniebroke) [reply src/auto_optional/code_processing.py:144]: I can do that tomorrow. I've allowed edits of this PR by maintainers, if you'd rather have it fixed today.

**Review by Luttik — APPROVED** [review]

**Comment by Luttik** [comment]
A new release has been added using this improvement.

**Pull request merged**
